### PR TITLE
feat: split OpenSearch from Elasticsearch as separate connection type

### LINKED ISSUE #423: [Refactor] Split OpenSearch from Elasticsearch as a separate connection type
## Problem

OpenSearch and Elasticsearch currently share a single `ElasticsearchConnection` type, distinguished only by a boolean `isOpenSearch` flag detected at runtime during `freshConnection()`. This creates several problems:

- **No type-level discrimination**: `DatabaseType.OPENSEARCH` doesn't exist — both are `DatabaseType.ELASTICSEARCH` with `isOpenSearch: true/false`
- **Boolean drift risk**: The `isOpenSearch` flag is spread across 6+ locations in `connectionStore.ts` and `esApi.ts`
- **Product confusion**: Users can't explicitly create an OpenSearch connection — it's auto-detected from cluster info, which is fragile
- **Blocks EasySearch integration**: The current "one type for all search engines" pattern would get worse with a third variant

## Solution

Refactor the type hierarchy to give each search engine its own discriminated type.

### Type Changes (src/store/connectionStore.ts)

```typescript
export enum DatabaseType {
  ELASTICSEARCH = 'ELASTICSEARCH',
  OPENSEARCH = 'OPENSEARCH',
  DYNAMODB = 'DYNAMODB',
  MONGODB = 'MONGODB',
}

// Shared base for HTTP-API-compatible search engines
export type SearchConnectionBase = {
  id?: number;
  name: string;
  indices: Array<ElasticSearchIndex>;
  host: string;
  port: number;
  authType?: 'basic' | 'apiKey';
  username?: string;
  sslCertVerification: boolean;
  password?: string;
  apiKey?: string;
  queryParameters?: string;
  activeIndex: ElasticSearchIndex | undefined;
  version: string;
  clusterName: string;
  clusterUuid: string;
};

export type ElasticsearchConnection = SearchConnectionBase & {
  type: DatabaseType.ELASTICSEARCH;
};

export type OpenSearchConnection = SearchConnectionBase & {
  type: DatabaseType.OPENSEARCH;
};

export type SearchConnection = ElasticsearchConnection | OpenSearchConnection;

// Type guards
export const isSearchConnection = (connection: Connection): connection is SearchConnection => { ... };
export const isOpenSearchConnection = (connection: Connection): connection is OpenSearchConnection => { ... };
```

### Schema Migration

Bump `SCHEMA_VERSION` to 5. Add V4->V5 migration that:
1. Normalizes persisted `type` values to uppercase
2. Converts `{ type: ELASTICSEARCH, isOpenSearch: true }` to `{ type: OPENSEARCH }`, stripping `isOpenSearch`
3. Takes a backup before migrating (`connections_v4_backup`)

### Replace all `connection.isOpenSearch`

Replace `connection.isOpenSearch` with `connection.type === DatabaseType.OPENSEARCH` in:
- `src/store/connectionStore.ts` — `freshConnection`, `fetchIndices`
- `src/datasources/esApi.ts` — `getTemplateApiMode`, `catIndices`

### Replace broad ELASTICSEARCH checks with `isSearchConnection()`

Where the intent is "is this a search engine (not DynamoDB/MongoDB)", replace `connection.type === DatabaseType.ELASTICSEARCH` with `isSearchConnection(connection)`.

**Core stores:**
- `src/store/connectionStore.ts` — `fetchIndices`, `searchQDSL`, `queryToCurl`
- `src/store/clusterManageStore.ts` — 17 ELASTICSEARCH checks
- `src/store/tabStore.ts` — panel routing
- `src/store/importExportStore.ts` — 12+ import/export branches

**DatabaseSource:**
- `src/datasources/esApi.ts` — change param types from `ElasticsearchConnection` to `SearchConnection`

**Monaco/Editor:**
- `src/views/editor/es-editor/index.vue` — engineType detection, history entries
- `src/common/monaco/referDoc.ts` — doc backend selection
- `src/common/monaco/completion.ts` — completion backend

**Connect UI:**
- `src/views/connect/components/connect-list.vue` — labels, icons, version, auth badges
- `src/views/connect/components/floating-menu.vue` — add OpenSearch to type picker
- `src/views/connect/index.vue` — panel routing
- `src/views/connect/components/es-connect-dialog.vue` — parameterize by SearchDatabaseType

**Toolbar/Manage:**
- `src/components/tool-bar.vue`
- `src/views/manage/index.vue`
- `src/views/manage/components/cluster-state.vue`

**Import/Export UI:**
- `src/views/import-export/components/export-source-scope.vue`
- `src/views/import-export/components/import-schema-structure.vue`
- `src/views/import-export/components/import-source-scope.vue`
- `src/views/import-export/components/import-target-output.vue`

**History:**
- `src/views/history/index.vue`

### Connect Dialog

Refactor `es-connect-dialog.vue` to accept a `SearchDatabaseType` parameter instead of hardcoding Elasticsearch. The host/port/auth form fields are identical for all search engines.

## Effort

**Medium** (~3-5 days). The refactoring is mechanical but touches ~30+ files.

## Risks

1. **Missing a broad ELASTICSEARCH check** — OpenSearch connections fail silently. Mitigation: thorough QA with an OpenSearch cluster.
2. **`isOpenSearch` replacement miss** — OpenSearch-specific behavior breaks. Mitigation: grep all 6 sites, test template API mode + expand_wildcards.
3. **Migration failure** — Corrupted saved connections. Mitigation: backup-then-migrate pattern.

### PR DESCRIPTION
**Pull request by AnnChord**
## Summary

This PR implements GitHub issue #423 by refactoring the type hierarchy to give each search engine its own discriminated type, splitting OpenSearch from Elasticsearch as a separate connection type.

#### Type System Changes
- Added `OPENSEARCH` to `DatabaseType` enum
- Created `SearchConnectionBase` type as shared base for HTTP-API-compatible search engines
- Created `OpenSearchConnection` type and `SearchConnection` union type
- Added type guards: `isSearchConnection()` and `isOpenSearchConnection()` with null/undefined handling

#### Schema Migration
- Bumped `CONNECTION_SCHEMA_VERSION` to 5 (renamed from `SCHEMA_VERSION`)
- Created consolidated `migrateConnections()` method for all version migrations (V1→V2, V2→V3, V4→V5)
- All migrations are **idempotent** for safe re-execution if schemaVersion write fails
- Robust error handling: backup/migration/cleanup failures handled gracefully

#### UI Updates
- OpenSearch selection properly initializes dialog with correct type
- Renamed misleading names to search-oriented: `isSearchConnectionComputed`, `activeSearchIndexOption`
- Connection persistence includes refreshed metadata (type, clusterName, clusterUuid)

#### Test Coverage
- Added tests for `migrateConnections` covering all transitions
- Added tests for idempotent migrations
- Added tests for `fetchConnections` edge cases
- Fixed test type declarations
- Coverage: connectionStore.ts at 55.59%

### Verification
- Lint: ✅ Passes
- Tests: ✅ 1053 tests pass
- Build: ✅ Succeeds

### Review Comments Addressed
All 7 Copilot review comments addressed in commits 208719f and c43c82c.

Closes #423

### CODE REVIEW

File: src/store/connectionStore.ts [hunk 1]
```diff
@@ -428,38 +510,49 @@ export const useConnectionStore = defineStore('connectionStore', {
...
             migrated = result.migrated;
             consolidatedCount = result.consolidatedCount;
             originalCount = result.originalCount;
-          }
 
-          if (storedVersion < 3) {
-            migrated = migrateDynamoConnectionsV2ToV3(migrated);
-          }
+            await storeApi.set('connections', pureObject(migrated));
+            await storeApi.set('schemaVersion', CONNECTION_SCHEMA_VERSION);
```
[the remark this answers is not in the document]
Comment (AnnChord) [reply src/store/connectionStore.ts:545]: Addressed in commit 208719f: Made all migrations idempotent to handle safe re-execution. V1→V2 now detects V2 connections (with auth field) and passes them through unchanged. V2→V3 and V4→V5 were already idempotent (check for presence of auth/isOpenSearch fields). Now if schemaVersion write fails after connections write, re-running migrations will not corrupt data.

**Comment by AnnChord** [comment]
### Review Comments Addressed

**Comment: V4→V5 migration tests** (3238894211)
- Addressed: Tests were consolidated into `migrateConnections` test block
- Covers: `isOpenSearch: true` → OPENSEARCH, `isOpenSearch: false` → ELASTICSEARCH, non-search unchanged, mixed connections

**Comment: Atomic writes / re-execution safety** (3245861794)  
- Addressed in commit 208719f: Made all migrations **idempotent**
- V1→V2 now detects V2 connections (with `auth` field) and passes through unchanged
- V2→V3 and V4→V5 were already idempotent (check for field presence before processing)
- Safe re-execution: if schemaVersion write fails, re-running migrations won't corrupt data

**Codecov: Patch coverage 72.88%**
- Coverage improved from 47.13% → 55.59%
- Added tests for idempotent migrations
- Added tests for fetchConnections edge cases (backup failure, migration failure, cleanup failure)

**Comment by AnnChord** [comment]
### All Copilot Review Comments Addressed

| ID | Issue | Resolution | Commit |
|----|-------|------------|--------|
| 3238894211 | V4→V5 tests | ✅ Tests consolidated in migrateConnections block | 6f3e473, 93e1cd2 |
| 3238894229 | type persistence | ✅ Already fixed - Object.assign includes type, clusterName, clusterUuid | Previous |
| 3245861763 | OpenSearch selection ignored | ✅ Fixed - showMedal now accepts initialType parameter | c43c82c |
| 3245861783 | storeApi type declaration | ✅ Fixed - type now includes get and delete | c43c82c |
| 3245861794 | Atomic writes | ✅ Fixed - all migrations now idempotent | 208719f |
| 3245861799 | isElasticsearchConnection misleading | ✅ Fixed - renamed to isSearchConnectionComputed | c43c82c |
| 3245861806 | activeElasticsearchIndexOption misleading | ✅ Fixed - renamed to activeSearchIndexOption | c43c82c |

### Summary
All 7 Copilot review comments have been addressed. The PR now has:
- Idempotent migrations for safe re-execution
- Proper OpenSearch initialization in connect dialog
- Search-oriented naming (isSearchConnectionComputed, activeSearchIndexOption)
- Correct test type declarations

**Verification:**
- 1053 tests pass
- Lint clean
- Build succeeds

**Review by Blankll — APPROVED** [review]
looks good
